### PR TITLE
Fix segmentations

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR reruns the IBP after fixing segmentation by performing seeded watershed for the cell segmentation algorithm

### CODE REVIEW

File: 3c.segment_cells/HPC_run_segmentation_child.sh [hunk 1]
```diff
@@ -18,13 +18,17 @@ conda activate timelapse_segmentation_env
...
 
-cd scripts/ || exit
+cd notebooks/ || exit
 
 
-python 2.nuclei_segmentation.py --input_dir "$main_dir" --clip_limit 0.6 --diameter 70
-python 2.nuclei_segmentation.py --input_dir "$terminal_dir" --clip_limit 0.6 --diameter 70
-
-python 3.cell_segmentation.py --input_dir "$main_dir" --clip_limit 0.6 --diameter 100
+papermill 1.nuclei_segmentation.ipynb 2.nuclei_segmentation.ipynb \
```
Comment (wli51) [inline 3c.segment_cells/HPC_run_segmentation_child.sh]: Seems like here and below you are naming the output from running `1.xxx.ipynb` as `2.xxx.ipynb`, is there a reason for this numbering. If not, maybe a better naming out the output would help make things clearer.

File: 3c.segment_cells/scripts/1.nuclei_segmentation.py [hunk 2]
```diff
@@ -182,7 +123,9 @@ def get_masks(image, model, channels, diameter):
     (
         img,
         nuclei[img, :, :].shape,
-        model.eval(nuclei[img, :, :], channels=channels, diameter=diameter),
+        model.eval(
```
Comment (wli51) [inline 3c.segment_cells/scripts/1.nuclei_segmentation.py:126]: Consider standardizing the use of for loop vs list comprehension here. Seems like you can use a single for loop on nuclei.shape[0] to handle the normalization, model eval and updating to `masks_all_dict`

File: 4.cell_tracking/scripts/0.nuclei_tracking.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def clear_gpu_memory():
 
 else:
     print("Running in a notebook")
-    well_fov = "C-02_F0001"  # example well_fov
+    well_fov = "C-02_F0003"  # example well_fov
```
Comment (wli51) [inline 4.cell_tracking/scripts/0.nuclei_tracking.py:79]: Seems like the changes here only involves changing a notebook parameter and re-running the entire thing.  If you need to re-run the same workflow on different wells a lot, maybe the best way is not to use a notebook and instead keep the same script parsing arguments and logging run specific outputs to a location of log files and plots? This will reduce the size of future PRs.

**Review by wli51 — APPROVED** [review]
PR LGTM. 

I left comments about 1 nitpick about loop and list comprehension consistency.
When reviewing the PR I also have the overall impression that this will be an analysis to be ran on different data every time, and a lot of that is being done with changing one or a few parameters and re-running the notebook. Doing too much of this will make the PRs very big and hard to track down by a single reviewer. A good future direction would be making some of the re-suable analysis scripts with arguments.

File: 3c.segment_cells/HPC_run_segmentation_child.sh (hunk 1 above)
Comment (MikeLippincott) [reply 3c.segment_cells/HPC_run_segmentation_child.sh]: adjusted to write notebooke 1 to notebook 1

File: 4.cell_tracking/scripts/0.nuclei_tracking.py (hunk 3 above)
Comment (MikeLippincott) [reply 4.cell_tracking/scripts/0.nuclei_tracking.py:79]: Good point! I usually run a bash script that iterates over multiple injectable parameters for the nbconverted script :)

File: 3c.segment_cells/scripts/1.nuclei_segmentation.py (hunk 2 above)
Comment (MikeLippincott) [reply 3c.segment_cells/scripts/1.nuclei_segmentation.py:126]: Good point too. There is a really weird bug with CellPose and the python interpreter in the notebooks vs script where a memory leak occurs and this weird combination is the only way to avoid it... :/
